### ISSUE TITLE
Flow type causes `object-curly-newline` false positive with ESLint 4

### ISSUE THREAD

**Issue by nwoltman** [babel/babel-eslint#513]
Quick link: [`object-curly-newline`](https://eslint.org/docs/rules/object-curly-newline)

#### ESLint config

```json
{
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module",
    "ecmaVersion": 2017,
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "rules": {
    "object-curly-newline": ["error", {"consistent": true}]
  }
}
```

#### Code
```jsx
// @flow

type Params = {
  a: number,
  b: number,
};

export function sum({ // Error happens here
  a,
  b,
}: Params) {
  return a + b;
}
```

#### Error
```
8:31  error  Unexpected line break after this opening brace  object-curly-newline
```

The error goes away if I remove the `: Params` type annotation.

eslint version: 4.5.0
babel-eslint version: 7.2.3/8.0.0-alpha.17

I did not have this error with ESLint v3.19.0.

**Comment by Whoaa512** [comment]
Also seeing this but my error says `Unexpected line break before this opening brace  object-curly-newline`

**Comment by pascalduez** [comment]
Hi,

If I format as follow:
```js
// @flow

type Params = {
  a: number,
  b: number,
};

export function sum({
  a,
  b,
}:
Params) {
  return a + b;
}
```

Which I would prefer not to be forced to, then the `eslint-plugin-flowtype` error, for good reasons.

```
error  There must not be a line break after "{
  a,
  b,
}" parameter type annotation colon  flowtype/space-after-type-colon
```

**Comment by nwoltman** [comment]
Fixed in ESLint v4.11.0
Thanks @TiddoLangerak!
